### PR TITLE
Add class attributes explicitly in wrapper classes

### PR DESCRIPTION
**Pull request by zundertj**

### CODE REVIEW

File: py-polars/polars/internals/lazy_frame.py [hunk 1]
```diff
@@ -1071,8 +1070,10 @@ class LazyGroupBy:
     Created by `df.lazy().groupby("foo)"`
     """
 
+    _lgb: "PyLazyGroupBy"
```
Comment (zundertj) [inline py-polars/polars/internals/lazy_frame.py:1073]: Renamed from `lgb` to `_lgb`, we dont want this part of the API, do we?

Comment (ritchie46) [reply py-polars/polars/internals/lazy_frame.py:1073]: Yes, its private. :+1:

**Comment by ritchie46** [comment]
Help me with the rationale of this one. Why not on the object instance?

**Comment by zundertj** [comment]
We always bind on the object instance in the `__init__`, so that would have prio over class attribute. But it may be confusing indeed.

**Pull request closed without merging**